src/Common/BackgroundShellCommandHolder.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include <memory>
+#include <mutex>
+
+#include <Common/ShellCommand.h>
+#include <unordered_map>
+
+
+namespace DB
+{
+
+/** The holder class for running background shell processes.
+*/
+class BackgroundShellCommandHolder final
```

Comment:
Let's mark this class non-copyable and non-movable explicitly (`std::mutex` makes this implicitly). For example, by inheriting  from `boost::noncopyable`